README.markdown

```diff
@@ -95,6 +95,65 @@ end
 
 See `lib/fakefs/spec_helpers.rb` for more info.
 
+**NOTE**: There is a bug currently, that causes FakeFS to fail with the following error:
+```
+TypeError:
+  superclass mismatch for class File
+```
+
+To prevent this, you'll need to require 'pp' *BEFORE* you require 'fakefs/spec_helpers' in your
+spec_helpers file. Like this:
+
+```ruby
+require 'pp'
+require 'fakefs/spec_helpers'
+
+describe "my spec" do
+  include FakeFS::SpecHelpers
+end
+```
+
+Mocking relative path in a Gem
+------------------------------
+
+If you are working in a Gem and would like to load configuration files which sit with your gem in a
+cfg folder, you will need to clone your filesystem first. To do this, add the following line to your
+spec_helpers file.
+
+```ruby
+...
+shared_examples 'with faked out filesystem' do
+  before :each do
+    include FakeFS::SpecHelpers
+    FakeFS.activate!
+    FakeFS::FileSystem.clone(__dir__)
+  end
+end
+...
+```
+
+And than if you have a test which needs to use a faked filesystem, like a configuration test, you'll
+have to use this setup in the test like this.
+
+```ruby
+...
+describe MyModule::MySuperAwesomeClass, fakefs: true do
+  include_context 'with faked out filesystem'
+  ...
+```
+
+Once this is done, your configuration class will now look in the faked out filesystem, so you'll have
+to create whatever configuration file you have in a setup, like this.
+
+```ruby
+FakeFS do
+  FileUtils.mkdir_p(File.join(__dir__, '../cfg'))
```

Comment:
Right. I will re-word this part. I was trying to say, that, if you would like to overwrite the existing one, because you are entering test data. Like, if you have a public key file there, but you don't want to go through the hassle of it, just enter asdf.